Lib/test/test_pstats.py

```diff
@@ -30,12 +34,39 @@ class StatsTestCase(unittest.TestCase):
     def setUp(self):
         stats_file = support.findfile('pstats.pck')
         self.stats = pstats.Stats(stats_file)
+        to_compile = 'import os'
+        self.temp_storage = tempfile.NamedTemporaryFile(delete=False)
+        profiled = compile(to_compile, '<string>', 'exec')
+        cProfile.run(profiled, filename=self.temp_storage.name)
+
+    def tearDown(self):
+        self.temp_storage.close()
+        os.remove(self.temp_storage.name)
 
     def test_add(self):
         stream = StringIO()
         stats = pstats.Stats(stream=stream)
         stats.add(self.stats, self.stats)
 
+    def test_dump_and_load_works_correctly(self):
+        temp_storage_new = tempfile.NamedTemporaryFile(delete=False)
+        self.stats.dump_stats(filename=temp_storage_new.name)
+        tmp_stats = pstats.Stats(temp_storage_new.name)
+        self.assertEqual(self.stats.stats, tmp_stats.stats)
+        temp_storage_new.close()
+        os.remove(temp_storage_new.name)
+
+    def test_load_equivalent_to_init(self):
+        empty = pstats.Stats()
+        empty.load_stats(self.temp_storage.name)
```

Comment:
Could you rename the variable `empty` as it's not empty?

Reply:
modified.